src/server/json_family.cc

```diff
@@ -84,6 +84,16 @@ string JsonTypeToName(const JsonType& val) {
   return std::string{};
 }
 
+JsonExpression ParseJsonPath(string_view path, error_code* ec) {
```

Comment:
```suggestion
JsonExpression ParseJsonPath(string_view path, error_code& ec) {
```

Reply:
The reason I like `ref` instead of  a `pointer` is that the second can be used to denote an optional value (meaning that it can be a `nullptr`) -- that's from Sean Parent suggestion. Also bonus point, you don't need to dereference :)

Reply:
I agree, and I also raised this when I first started contributing to this project.
The reason why we take pointers is due to the fact that the old Google Style Guide (which our project follows) used to only allow either `const&` or mutable pointers. There are some pros to that, but it doesn't matter because Google switched away from it. But since our code base was developed with this in mind, @romange said we should continue with this.

Reply:
That makes sense. @dranikpg ignore me then.

Reply:
also for completion do we have a reference to the version of google style of code we use? 

Because from https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs

```
input/output parameters should usually be references (which cannot be null). 
```

Reply:
I don't know. I remember Google making those changes a couple of years ago, when I still worked there :)
I can't even find any place that archives the old versions, although admittedly I haven't looked too hard